### DISCUSSION TITLE
How to pass workflow metadata labels to pod labels or annotations

### DISCUSSION THREAD

**Discussion by dpeer6** [argoproj/argo-workflows#6321] [General]
Hi,

I would like to pass `creator` label, which automatically added to any workflow, to the workflow pods, so I can later capture/filter it with our logging system.

Example from one of my workflows:
Labels:
     workflows.argoproj.io/creator=system-serviceaccount-argo-**dpeer**

I tried using `podMetadata` label/annotations in my workflow YAML but none worked.
Something like:
spec:
  podMetadata:
    annotations:
      myUser: "{{workflow.creator}}"

Do you have an idea how can I do that?

**Comment by dpeer6** [comment]
Can podSpecPatch do the trick?
If so, how?

**Reply by sarabala1979** [reply]
yes,  `podSpecPatch` will work for you scenario. you can try it.

**Reply by dpeer6** [reply]
Thank you... can you share how can i pass the creator label from the workflow to the pods?
I mean I tried to get that value, but could not find how. It appears only at the workflow object

**Comment by dpeer6** [comment]
There are several ways to pass info as pod label apparently, but how can I get/refer to that creator value from the workflow YAML?
"workflows.argoproj.io/creator=system-serviceaccount-argo-dpeer"

Using: {{workflow.labels.workflows.argoproj.io/creator}} does not work.

**Reply by sarabala1979** [reply]
Are you creating workflow through UI or CLI or API? 
To refer and resolve the passed value in it has to be workflow parameter (https://github.com/argoproj/argo-workflows/blob/fa4b32a5347d924079072fe760768462d40f6b3b/docs/variables.md). 
you have to pass it as a parameter

**Reply by dpeer6** [reply]
I am using the CLI.
I thought that there are some global variables I can refer to.
When I'm using `{{workflow.namespace}}` it does get passed as pod label/annotation to all pods in the workflow.

So I'm guessing I'm doing something wrong with that specific label.
(Yeh, I can use params, or some many other wrapping mechanism, but I like to use that specific label for the sake of simplicity and enforcement to really understand who's Bearer TOKEN was used)

I tried that too in several variations with no success:
{{workflow.labels['workflows.argoproj.io/creator']}}

**Comment by dpeer6** [comment]
ok, resolved.

I might had a typo or something similar, but this worked for me:
```yaml
podMetadata:
    labels:
      myUser: "{{workflow.labels.workflows.argoproj.io/creator}}"
```
